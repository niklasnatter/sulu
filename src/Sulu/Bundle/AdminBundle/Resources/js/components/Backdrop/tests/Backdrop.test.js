// @flow
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Backdrop from '../Backdrop';

test('The component should render', () => {
    const {container} = render(<Backdrop />);
    expect(container).toMatchSnapshot();
});

test('The component should call a function when clicked', async() => {
    const onClickSpy = jest.fn();
    render(<Backdrop onClick={onClickSpy} />);
    const backdrop = screen.queryByTestId('backdrop');

    expect(onClickSpy).toHaveBeenCalledTimes(0);
    await userEvent.click(backdrop);
    expect(onClickSpy).toHaveBeenCalledTimes(1);
});
