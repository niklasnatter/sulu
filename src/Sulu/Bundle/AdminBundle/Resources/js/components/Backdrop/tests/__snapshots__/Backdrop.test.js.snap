// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render 1`] = `
<div
  class="backdrop visible fixed"
  data-testid="backdrop"
  role="button"
/>
`;
